akd/agents/gap_analysis/prompts.py

```diff
@@ -0,0 +1,259 @@
+
+Output Example:
+[['KnowledgeHub: An End-to-End Tool for Assisted Scientific Discovery'], ['Abstract'], ['1 Introduction'], ['2 System Description', '2.1 Document Ingestion', '2.2 Annotation', '2.3 Question Answering'], ['3 Use-case: Knowledge Discovery for the Battery Domain'], ['4 Conclusion'], ['Ethical Statement'], ['Acknowledgments'], ['References']]
+
+Make sure to preserve the hierarchy of sections and subsections. ONLY provide the output.
+Now, group the following lists in the same way."""
+
+
+section_grouper_prompt = ChatPromptTemplate.from_messages(
+    [
+        ("system", section_grouper_inst),
+        ("user", "{input_sections}"),
+    ],
+)
+
+
+# =============================================================================
+# Classifies research paper section titles into predefined key categories like
+# introduction, methodology, and conclusion.
+# =============================================================================
+
+section_classifier_inst = """You are an expert in text categorization. Your task is to group section titles from research papers into one of the following key sections:
+
+## Key Sections
+`['introduction', 'related work or background', 'methodology', 'experiments, models and datasets', 'results and discussions', 'limitations', 'future work', 'conclusion', 'appendix', 'misc']`
+
+---
+
+## Instructions
+1. Assign each section title to the most appropriate key section based on its content and context.
+2. If a section title does not clearly fit into one of the predefined key sections, group it under **`misc`**.
+3. Sometimes, there are sections between introduction/related work and experiments/discuss that may be related to methodology. Group accordingly.
+4. If there are no sections under a particular key, leave the list empty.
+5. Use the following example mappings as a guide to your decisions:
+
+---
+
+## Input
+[List of section titles]
+
+### Example Input
+['Deep Convolutional Neural Networks for Palm Fruit Maturity Classification *', '1 Introduction', '2 Related Work', '3 Proposed Method', 'Background', '4 Experiments', '5 Limitations', '6 Discussion and Conclusion', '7 References']
+
+---
+
+## Output
+A dictionary where each key is a key section, and the value is a list of section titles grouped under that key.
+Only provide a JSON dictionary as output. Do not include any explanation or text outside of the dictionary.
+
+
+### Example Output
+'introduction': ['1 Introduction'],
+'related work or background': ['2 Related Work', 'Background'],
+'methodology': ['3 Proposed Method'],
+'experiments, models and datasets': ['4 Experiments'],
+'results and discussions': ['6 Discussion and Conclusion'],
+'limitations': ['5 Limitations'],
+'future work': [],
+'conclusion': ['6 Discussion and Conclusion'],
+'appendix': [],
+'misc': ['Deep Convolutional Neural Networks for Palm Fruit Maturity Classification *', '7 References']
+"""
+
+section_classifier_prompt = ChatPromptTemplate.from_messages(
```

Comment:
Same:
Can we make this all CAPSLOCK to conform to global variable naming convention?